app/api/common.v2/contracts/SyncDBDataSource.ts

```diff
@@ -0,0 +1,9 @@
+import { DeleteResult } from 'mongodb';
+
+export interface SyncDBDataSource<T> {
```

Comment:
I think this should go to the database/infrastructure directory. It is only database related, and also knows about MongoDB :)